ubg/src/mafia/MafiaNight.js

```diff
@@ -30,44 +30,101 @@ const useStyles = makeStyles((theme) => ({
 export default function MafiaNight({user, usersData, room,
   mafiaKill, doctorSave, detectiveCheck}) {
   const classes = useStyles();
+  const [initialize, setInitialize] = React.useState(false);
   const [players, setPlayers] = React.useState([]);
   const [userInfo, setUserInfo] = React.useState('');
   const [roleText, setRoleText] = React.useState('');
+  const [message, setMessage] = React.useState('');
 
   /**
    * @return {undefined}
    */
   function loadNightData() {
-    const allPlayers = [];
-    usersData.forEach(function(u) {
-      if (u.alive === true) {
-        allPlayers.push(u);
-      }
-      if (u.uid === user.uid) {
-        setUserInfo(u);
-        if ( u.role=== 1) {
-          setRoleText('Pretend to be clicking or thinking :)');
-        } else if (u.role === 2) {
-          setRoleText('Mafia, pick someone to kill.');
-        } else if (u.role === 3) {
-          setRoleText('Detective, who do you want to check tonight?');
-        } else if (u.role === 4) {
-          setRoleText('Doctor, who do you want to save tonight?');
+    if (mafiaKill['uid'] !== '' &&
+    doctorSave['uid'] !== '' &&
+    detectiveCheck['uid'] !== '') {
+      room.update({day: true});
+    }
+    if (initialize === false) {
+      setInitialize(true);
+      const roles = new Set();
+      const allPlayers = [];
+      usersData.forEach(function(u) {
+        if (u.alive === true) {
+          roles.add(u.role);
+          allPlayers.push(u);
+        }
+        if (u.uid === user.uid) {
+          setUserInfo(u);
+          if ( u.role=== 1) {
+            setRoleText('Pretend to be clicking, tapping or thinking :)');
+          } else if (u.role === 2) {
+            setRoleText('Mafia, pick someone to kill.');
+          } else if (u.role === 3) {
+            setRoleText('Detective, who do you want to check tonight?');
+          } else if (u.role === 4) {
+            setRoleText('Doctor, who do you want to save tonight?');
+          }
+        }
+      });
+      // This is probably not best practice, tell me if you have better idea
+      // I think this should probably be done in Room.js?
+      for (let i = 2; i < 5; i++) {
```

Comment:
What are you trying to do here? Initialize mafiaKill, detectiveCheck and doctorSave?

Reply:
I'll leave night when uid of mafiaKill, detectiveCheck and doctorSave are not ''. But there is the possibility that there is no doctor/detective/mafia. So I'll set the uid of the role that is not filled with '#'.

Reply:
Maybe then do it in the initialize if, or do you need to run this every time useEffect is called?